src/MakersOfDenmark.Application/Queries/V1/GetSelectionOfMakerSpaces.cs

```diff
@@ -0,0 +1,57 @@
+﻿using MakersOfDenmark.Domain.Enums;
+using MakersOfDenmark.Domain.Models;
+using MakersOfDenmark.Infrastructure.Persistence;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MakersOfDenmark.Application.Queries.V1
+{
+    public class GetSelectionOfMakerSpaces : IRequest<GetSelectionOfMakerSpacesResponse>
+    {
+        public GetSelectionOfMakerSpaces(string name)
+        {
+            Name = name;
+        }
+
+        public string Name { get; }
+    }
+    
+    public class GetSelectionOfMakerSpacesHandler : IRequestHandler<GetSelectionOfMakerSpaces, GetSelectionOfMakerSpacesResponse>
+    {
+        private readonly MODContext _context;
+
+        public GetSelectionOfMakerSpacesHandler(MODContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<GetSelectionOfMakerSpacesResponse> Handle(GetSelectionOfMakerSpaces request, CancellationToken cancellationToken = default)
+        {
+            var makerSpace = await _context.MakerSpace.Include(x => x.Address).AsNoTracking().FirstAsync(x => x.Name == request.Name);
```

Comment:
God brug af AsNoTracking(). Det gør din query mere performant, hvis du altså ikke skal lave nogle ændringer.

Reply:
Når man bruger FirstAsync skal man passe på - hvis den ikke finder noget i listen, så thrower den en exception.

Ved brug af FirstOrDefaultAsync() er det i stedet default værdien af MakerSpace der bliver returneret, hvis intet kan findes, hvilket i dette tilfælde er null

https://stackoverflow.com/questions/1024559/when-to-use-first-and-when-to-use-firstordefault-with-linq